tests/model/test_scvi.py

```diff
@@ -475,12 +475,20 @@ def test_scvi_n_obs_error(n_latent: int = 5):
     adata = adata[0:129].copy()
     SCVI.setup_anndata(adata)
     model = SCVI(adata, n_latent=n_latent)
-    with pytest.raises(ValueError):
+    with pytest.warns(UserWarning):
         model.train(1, train_size=1.0)
```

Comment:
train_size=1.0 should lead to exactly this train size. There needs to be an option to fix train_size. None could lead to train_size 0.9 if possible and otherwise adjust it.